2.segment_images/scripts/6.post-hoc_reassignment.py

```diff
@@ -0,0 +1,354 @@
+    Parameters
+    ----------
+    mask : np.ndarray
+        The mask to process, should be a 3D numpy array
+    border : int, optional
+        The number of pixels in width to create border to scan for edge cased, by default 10
+
+    Returns
+    -------
+    np.ndarray
+        The mask with edge cases removed
+    """
+
+    edge_pixels = np.concatenate(
+        [
+            # all of z, last n rows (y), all columns (x) - bottom edge
+            mask[:, -border:, :].flatten(),
+            # all of z, first n rows (y), all columns (x) - top edge
+            mask[:, 0:border, :].flatten(),
+            # all of z, all rows (y), first n columns (x) - left edge
+            mask[:, :, 0:border:].flatten(),
+            # all of z, all rows (y), last n columns (x) - right edge
+            mask[:, :, -border:].flatten(),
+            # each are the edges stacked for the whole volume -> no need to specify every z slice or 3D edge
+        ]
+    )
+    # get unique edge pixel values
+    edge_pixels = np.unique(edge_pixels[edge_pixels > 0])
+
+    for edge_pixel_case in edge_pixels:
+        # make the edge cases equal to zero
+        mask[mask == edge_pixel_case] = 0
+
+    # return the mask with edge cases removed
+    return mask
+
+
+def centroid_within_bbox_detection(
+    centroid: tuple,
+    bbox: tuple,
+) -> bool:
+    """
+    Check if the centroid is within the bbox
+
+    Parameters
+    ----------
+    centroid : tuple
+        Centroid of the object in the order of (z, y, x)
+        Order of the centroid is important
+    bbox : tuple
+        Where the bbox is in the order of (z_min, y_min, x_min, z_max, y_max, x_max)
+        Order of the bbox is important
+
+    Returns
+    -------
+    bool
+        True if the centroid is within the bbox, False otherwise
+    """
+    z_min, y_min, x_min, z_max, y_max, x_max = bbox
+    z, y, x = centroid
+    # check if the centroid is within the bbox
+    if (
+        z >= z_min
+        and z <= z_max
+        and y >= y_min
+        and y <= y_max
+        and x >= x_min
+        and x <= x_max
+    ):
+        return True
+    else:
+        return False
+
+
+def check_if_centroid_within_mask(
+    centroid: tuple, mask: np.ndarray, label: int
+) -> bool:
+    """
+    Check if the centroid is within the mask
+
+    Parameters
+    ----------
+    centroid : tuple
+        Centroid of the object in the order of (z, y, x)
+        Order of the centroid is important
+    mask : np.ndarray
+        The mask to check against
+
+    Returns
+    -------
+    bool
+        True if the centroid is within the mask, False otherwise
+    """
+    z, y, x = centroid
+    z = np.round(z).astype(int)
+    y = np.round(y).astype(int)
+    x = np.round(x).astype(int)
+    # check if the centroid is within the segmentation mask
+    cell_label = mask[z, y, x]
+    if cell_label > 0 and cell_label == label:
+        return True
+    else:
+        return False
+
+
+def mask_label_reassignment(
```

Comment:
Consider moving the functions found within these notebooks to a more structured Python package (directory) of modules (`.py`) files so they may be imported for use within the notebooks. This would lend itself well to reuse and testing patterns which benefit programming velocity, validation, and organization.